rate-limiter/src/main/scala/it/pagopa/interop/commons/ratelimiter/akkahttp/RateLimiterDirective.scala

```diff
@@ -0,0 +1,45 @@
+package it.pagopa.interop.commons.ratelimiter.akkahttp
+
+import akka.http.scaladsl.marshalling.ToEntityMarshaller
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directive1
+import akka.http.scaladsl.server.Directives.{onComplete, provide}
+import akka.http.scaladsl.server.directives.RouteDirectives._
+import com.typesafe.scalalogging.LoggerTakingImplicit
+import it.pagopa.interop.commons.logging.ContextFieldsToLog
+import it.pagopa.interop.commons.ratelimiter.RateLimiter
+import it.pagopa.interop.commons.ratelimiter.akkahttp.Errors.MissingOrganizationIdClaim
+import it.pagopa.interop.commons.ratelimiter.error.Errors.TooManyRequests
+import it.pagopa.interop.commons.utils.AkkaUtils._
+import it.pagopa.interop.commons.utils.ORGANIZATION_ID_CLAIM
+import it.pagopa.interop.commons.utils.TypeConversions._
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+object RateLimiterDirective {
+
+  def rateLimiterDirective[T](rateLimiter: RateLimiter, tooManyRequestsProblem: T)(
```

Comment:
why was the directive made parametric in the problem? Doesn't make sense to have just one too many request error centralized in commons? (maybe I'm missing something)

Reply:
in the long run it definitely makes sense!
But right now that each service has its own `Problem`, this implementation allows to send whatever the service wants (with its marshaller).

If we want to address the topic I could start creating a `Problem` in `utils` module (which would require to include also `spray` dependency).
Not sure if this would have any side impact, I should check in case